### ISSUE TITLE
Logistic Regression

### ISSUE THREAD

**Issue by CodenameSource** [CodenameSource/libuaic#5]
Имплементиране на методите и функциите за логистична регресия

# Структура на логистичната регресия
- [x] Структура с коефициенти и указатели към методите на алгоритъма
- [x] Сигмоидна функция приемаща за вход резултат от линейна регресия
![image](https://github.com/CodenameSource/libuaic/assets/65411104/c5ff34a3-8878-451e-99fe-164523c0e0e6)
Сигмоидна функция
![image](https://github.com/CodenameSource/libuaic/assets/65411104/bd2fd2fa-0834-4dff-94cb-64d33b4cb507)
Енкапсулирана линейна регресия(В случая множествена линейна регресия)
- [x] Loss function, фунцкия калкулираща грешката на предсказания, от алгоритъма, резултат от реалния (За едно поле от тренировъчните данни)
![image](https://github.com/CodenameSource/libuaic/assets/65411104/31c17982-d931-46ee-8532-22ef6dfd0e6e)
->
![image](https://github.com/CodenameSource/libuaic/assets/65411104/51fdece9-95ae-4573-bf82-dea999de6c3f)

- [x] Cost function, функция калкулираща грешката на модела върху целите тренировъчни данни
![image](https://github.com/CodenameSource/libuaic/assets/65411104/7dd96b86-116d-467e-9705-74224b1cd5b6)

- [x] Optimizer function - Gradient descent, оптимизираща функция 
![image](https://github.com/CodenameSource/libuaic/assets/65411104/4640d92a-dc00-4094-bb0c-7e006e85cdaf)
Gradient descent, very scary math
- [x] Fit function, функция която тренира алгоритъма върху въведените данни
- [x] Predict function, функция която взима коефициентите и връща предсказания резултат спрямо въведените данни на базата на тренираните коефициенти

# Ресурси за справка:
- [Logistic regression from scratch](https://philippmuens.com/logistic-regression-from-scratch)
- [Gradient descent](https://philippmuens.com/gradient-descent-from-scratch)
- [Logistic regression from scratch(Medium)](https://towardsdatascience.com/coding-logistic-regression-from-scratch-d18b4fbfca8a)
- [Logistic regression intuition (Medium)](https://medium.com/analytics-vidhya/basics-and-beyond-logistic-regression-34549d2ee800)
